OpenRA.Game/InstalledMods.cs

```diff
@@ -69,26 +69,38 @@ Manifest LoadMod(string id, string path)
 			try
 			{
 				if (!Directory.Exists(path))
-					throw new InvalidDataException(path + " is not a valid mod package");
+				{
+					Log.Write("debug", path + " is not a valid mod package");
+					return null;
+				}
 
 				package = new Folder(path);
-				if (!package.Contains("mod.yaml"))
-					throw new InvalidDataException(path + " is not a valid mod package");
-
-				using (var stream = package.GetStream("icon.png"))
-					if (stream != null)
-						using (var bitmap = new Bitmap(stream))
-							icons[id] = sheetBuilder.Add(bitmap);
-
-				return new Manifest(id, package);
+				if (package.Contains("mod.yaml"))
```

Comment:
This change + move disposal to finally isn't sufficient, because we want disposal on error, missing mod.yaml, but on valid package so it would need more refactoring and the end result might be less concise than now.